src/protojure/internal/grpc/client/providers/http2/core.clj

```diff
@@ -53,16 +53,21 @@
   core.async channel closure as an error, and terminate the processing once the response contains
   the :status code.
   "
-  [meta-ch request]
+  [meta-ch]
   (p/promise
    (fn [resolve reject]
-     (go-loop [response {}]
-       (if-let [data (<! meta-ch)]
-         (let [response (merge response data)]
-           (if (contains? response :status)
-             (resolve response)
-             (recur response)))
-         (reject (ex-info "Unexpected disconnect receiving response" {:response response})))))))
+     (if-let [data (async/<!! meta-ch)]
+       (let [response (merge {} data)]
+         (when (contains? response :status)
+           (if (= (:status response) 200)
```

Comment:
I'm fairly certain 200 is the only status we'd continue with here, this needed to throw in order to get out of the p/all, but worth calling out as I wasn't sure if we ever need to handle something like a 301 at this level.

Reply:
I did this because the GRPC spec basically says it should always return a 200 and use the grpc-status/message headers to convey error status.  That said, your point on 301 handling is fair.  I'm not sure what the best thing to do here is, but I think its fair to say we can handle whatever that might be in a later patch ;)

Reply:
No, note there is a later check in the code still, which i think is what you recall writing. It's probably/possibly dead code now that i've not removed -- the gist of this new code is that the status equality with 200 check had to be moved up. I'll take a look at harvesting the original check.

Reply:
Ah, sorry, I misunderstood.  Use your judgement and I can review once you are satisfied.

Reply:
With the separation of the send & receive, the 200 check i had here was no longer necessary, removed. To your point, the receive-payload check is still in place.